cypress/integration/advanced-functionalities/admin/system/users/Delete.spec.js

```diff
@@ -0,0 +1,46 @@
+/// <reference types="cypress" />
+const adminURL = Cypress.env('ADMIN_URL')
+const email = Cypress.env('USER_EMAIL')
+const password = Cypress.env('USER_PASSWORD')
+
+describe('Test for deleting a user', () => {
+  before(() => {
+    if (!window.sessionStorage.getItem('auth.refresh-token')) {
+      cy.login({ email, password, url: adminURL })
+    }
+  })
+
+  context('Test for deleting a user', () => {
+    it('should be able to delete a user', () => {
+      cy.visit(adminURL + '/')
+      // We wait for 2s in order the page to be fully loaded
+      cy.wait(2000)
+      cy.get('.nav-sidebar').contains('Users').click()
+      // We wait 1s in order the page to be fully loaded
+      cy.wait(1000)
+      cy.contains('Permissions account').get('#resource-list > tbody > tr:last > td:last > a').click()
+      cy.get('[data-test-id="button-delete"]').click()
+      cy.get('.confirmation-confirm').click()
+      // We confirm that the action was completed successfully
+      cy.get('.b-toast-success')
+    })
+  })
+
+  context('Test for un-deleting a user', () => {
+    it('should be able to un-delete a user', () => {
+      cy.get('[data-test-id="filter-deleted-users"]').contains('Only').click()
+      // We wait 1s in order the page to be fully loaded
+      cy.wait(1000)
+      cy.get('[data-test-id="input-search"]').type('Permissions')   
+      // We wait 1s for the search to finish
+      cy.wait(1000)    
+      cy.get('#resource-list > tbody > tr:last > td:last > a').click()
+      cy.get('[data-test-id="input-deleted-at"]').should('exist')
+      cy.get('[data-test-id="button-undelete"]').click()
+      cy.get('.confirmation-confirm').click()
+      // We confirm that the action was completed successfully
+      cy.get('.b-toast-success')
+      cy.get('[data-test-id="input-deleted-at"]').should('not.exist')
+    })
+  })
```

Comment:
check if the un-deleted user doesn't have a deleted state + colour is as before + he doesn't show when filtering by deleted users